src/Microsoft.DotNet.Build.Tasks/GenerateTestExecutionScripts.cs

```diff
@@ -90,7 +90,7 @@ private void WriteShExecutionScript(string shExecutionTemplate, string outputPat
             foreach (string runCommand in TestCommands)
             {
                 testRunCommands.Append($"{runCommand}\n");
-                testRunEchoes.Append($"echo {runCommand}\n".Replace("(", "").Replace(")", ""));  // Remove parentheses from echo command to avoid errors on Linux
+                testRunEchoes.Append($"echo \"{runCommand}\"\n".Replace("(", "").Replace(")", ""));  // Remove parentheses from echo command to avoid errors on Linux
```

Comment:
I don't have an example where someone has already done this, but I can easily see where it'd go wrong.  Imagine if RunCommand contained the substring:

"  > " 

This would close the first double quote, tell it to pipe the output of that command into the thing started by the second double quote.